### ISSUE TITLE
Skupper 0.50 internal proxy isn't starting

### ISSUE THREAD

**Issue by keith-mcclellan** [skupperproject/skupper#459]
We have a cluster with 0.32 running, but we tried creating a fresh setup with 0.50 and are running into an issue where the internal proxy container isn't starting properly saying that it can't mount its secrets.  The secrets were definitely created though - any ideas? Is 0.50 not setting the permissions properly or something?

```
Keiths-MacBook-Pro:multi-cluster keith$ kubectl get secrets -n cockroachdb --context $(whoami)@$(whoami)-cockroachdb-skupper-aws.us-west-1.eksctl.io
NAME                                     TYPE                                  DATA   AGE
cockroachdb.client.root                  Opaque                                5      40m
cockroachdb.node                         Opaque                                5      40m
default-token-287w6                      kubernetes.io/service-account-token   3      40m
skupper-console-users                    Opaque                                1      40m
skupper-local-ca                         kubernetes.io/tls                     2      40m
skupper-local-client                     kubernetes.io/tls                     4      40m
skupper-local-server                     kubernetes.io/tls                     3      40m
skupper-router-token-kj69d               kubernetes.io/service-account-token   3      40m
skupper-service-controller-token-hq46k   kubernetes.io/service-account-token   3      40m
skupper-site-ca                          kubernetes.io/tls                     2      40m
skupper-site-server                      kubernetes.io/tls                     3      40m
```

```
Keiths-MacBook-Pro:multi-cluster keith$ kubectl get pods -n cockroachdb --context $(whoami)@$(whoami)-cockroachdb-skupper-aws.us-west-1.eksctl.io
NAME                                          READY   STATUS              RESTARTS   AGE
cluster-init-secure-bfvj4                     0/1     Completed           0          32m
cockroachdb-internal-west-proxy-0             0/1     ContainerCreating   0          32m
cockroachdb-west-0                            1/1     Running             0          33m
cockroachdb-west-1                            1/1     Running             0          33m
cockroachdb-west-2                            1/1     Running             0          33m
skupper-router-579fc6bfbf-x797z               1/1     Running             0          33m
skupper-service-controller-5cc7d6c6fb-sh9g6   1/1     Running             0          33m
```

```
Keiths-MacBook-Pro:multi-cluster keith$ kubectl describe pods cockroachdb-internal-west-proxy-0 -n cockroachdb --context $(whoami)@$(whoami)-cockroachdb-skupper-aws.us-west-1.eksctl.io
Name:           cockroachdb-internal-west-proxy-0
Namespace:      cockroachdb
Priority:       0
Node:           ip-192-168-87-73.us-west-1.compute.internal/192.168.87.73
Start Time:     Tue, 23 Mar 2021 09:24:16 -0400
Labels:         controller-revision-hash=cockroachdb-internal-west-proxy-5cb4776b54
                internal.skupper.io/service=cockroachdb-internal-west
                statefulset.kubernetes.io/pod-name=cockroachdb-internal-west-proxy-0
Annotations:    kubernetes.io/psp: eks.privileged
Status:         Pending
IP:
IPs:            <none>
Controlled By:  StatefulSet/cockroachdb-internal-west-proxy
Containers:
  proxy:
    Container ID:
    Image:          quay.io/skupper/qdrouterd:0.5.0
    Image ID:
    Port:           <none>
    Host Port:      <none>
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Environment:
      QDROUTERD_CONF:       [
                                [
                                    "router",
                                    {
                                        "id": "$HOSTNAME",
                                        "mode": "edge",
                                        "metadata": "{\"id\":\"b439faac-30b9-48fa-8be5-1e5f613fdc3d\",\"version\":\"0.5.0\"}"
                                    }
                                ],
                                [
                                    "sslProfile",
                                    {
                                        "name": "skupper-internal",
                                        "certFile": "/etc/qpid-dispatch-certs/skupper-internal/tls.crt",
                                        "privateKeyFile": "/etc/qpid-dispatch-certs/skupper-internal/tls.key",
                                        "caCertFile": "/etc/qpid-dispatch-certs/skupper-internal/ca.crt"
                                    }
                                ],
                                [
                                    "connector",
                                    {
                                        "name": "uplink",
                                        "role": "edge",
                                        "host": "skupper-router.cockroachdb",
                                        "port": "45671",
                                        "sslProfile": "skupper-internal"
                                    }
                                ],
                                [
                                    "listener",
                                    {
                                        "name": "amqp",
                                        "host": "localhost",
                                        "port": 5672
                                    }
                                ],
                                [
                                    "tcpConnector",
                                    {
                                        "name": "egress",
                                        "host": "cockroachdb-west-${POD_ID}.cockroachdb-internal-west.cockroachdb",
                                        "port": "26257",
                                        "address": "cockroachdb-internal-west-${POD_ID}",
                                        "siteId": "b439faac-30b9-48fa-8be5-1e5f613fdc3d"
                                    }
                                ]
                            ]
      QDROUTERD_CONF_TYPE:  json
      NAMESPACE:            cockroachdb (v1:metadata.namespace)
    Mounts:
      /etc/qpid-dispatch-certs/skupper-internal/ from uplink (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from skupper-router-token-kj69d (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  uplink:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  skupper-internal
    Optional:    false
  skupper-router-token-kj69d:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  skupper-router-token-kj69d
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason       Age                  From               Message
  ----     ------       ----                 ----               -------
  Normal   Scheduled    32m                  default-scheduler  Successfully assigned cockroachdb/cockroachdb-internal-west-proxy-0 to ip-192-168-87-73.us-west-1.compute.internal
  Warning  FailedMount  14m (x2 over 16m)    kubelet            Unable to attach or mount volumes: unmounted volumes=[uplink], unattached volumes=[skupper-router-token-kj69d uplink]: timed out waiting for the condition
  Warning  FailedMount  9m42s (x8 over 30m)  kubelet            Unable to attach or mount volumes: unmounted volumes=[uplink], unattached volumes=[uplink skupper-router-token-kj69d]: timed out waiting for the condition
  Warning  FailedMount  99s (x23 over 32m)   kubelet            MountVolume.SetUp failed for volume "uplink" : secret "skupper-internal" not found
```

**Comment by grs** [comment]
I'm sorry, it looks like there is a bug in the 0.5.0 release preventing the correct setup of statefulset proxies for headless services. I will get a fix in shortly and we will aim to get a 0.5.1 update out shortly.

**Comment by grs** [comment]
Should be fixed now in 0.5.1
